py3status/modules/rate_counter.py

```diff
@@ -7,101 +7,157 @@
     config_file: file path to store the time already spent
         and restore it the next session
          (default '~/.i3/py3status/counter-config.save')
+    format: output format string
+        (default 'Time: {days} day {hours}:{mins:02d} Cost: {total}')
+    format_money: output format string
+        (default '{price}$')
     hour_price: your price per hour (default 30)
     tax: tax value (1.02 = 2%) (default 1.02)
 
+Format placeholders:
+    {days} The number of whole days in running timer
+    {hours} The remaining number of whole hours in running timer
+    {mins} The remaining number of whole minutes in running timer
+    {secs} The remaining number of seconds in running timer
+    {subtotal} The subtotal cost (time * rate)
+    {tax} The tax cost, based on the subtotal cost
+    {total} The total cost (subtotal + tax)
+    {total_hours} The total number of whole hours in running timer
+    {total_mins} The total number of whole minutes in running timer
+
+Money placeholders:
+    {price} numeric value of money
+
 Color options:
-    color_bad: Running
-    color_good: Stopped
+    color_running: Running, default color_good
+    color_stopped: Stopped, default color_bad
 
 @author Amaury Brisou <py3status AT puzzledge.org>
 """
+
+
 import os
 import time
 
 
+# No "magic numbers"
+SECS_IN_MIN = 60.0
+SECS_IN_HOUR = 60 * SECS_IN_MIN  # 3600
+SECS_IN_DAY = 24 * SECS_IN_HOUR  # 86400
+
+
 class Py3status:
     """
     """
     # available configuration parameters
     cache_timeout = 5
     config_file = '%s%s' % (os.environ['HOME'],
                             '/.i3/py3status/counter-config.save')
+    format = 'Time: {days} day {hours}:{mins:02d} Cost: {total}'
+    format_money = '{price}$'
     hour_price = 30
     tax = 1.02
 
     def __init__(self):
-        self.current_time = time.time()
-        self.diff_time = 0
-        self.full_text = ''
-        self.restarted = False
+        self.running = False
         self.saved_time = 0
-        self.start_time = time.time()
-        self.started = False
+        self.start_time = self.current_time
         try:
             # Use file to refer to the file object
             with open(self.config_file) as file:
                 self.saved_time = float(file.read())
         except:
             pass
 
-    def kill(self):
-        if self.started is True:
+    @property
+    def current_time(self):
+        """Get the current time.
+
+        Using a helper property to make it easy to keep consitency.
+        """
+        return time.time()
```

Comment:
I was thinking that having the property called `current_time` would make things look more uniform throughout the rest of the code, as well as making any future refactoring easier.

The previous implementation had `self.current_time = time.time()` and `self.start_time = time.time()` sprinkled throughout,

As well as:
`self.current_time = time.time()`
`self.diff_time = time.gmtime(self.current_time - self.start_time)`
and:
`self.saved_time = time.time() - self.start_time`

I thought that having:
`self.saved_time = self.current_time - self.start_time`
would be more consistent and readable.

I was also thinking that we could maybe transition to a different library, like `datetime`, so that we could just use the built-in `datetime.timedelta` instead of calculating it ourselves.  Having it defined in the one place would make it somewhat trivial.